### DISCUSSION TITLE
Defining PDE with float exponent

### DISCUSSION THREAD

**Discussion by Pranay-Nagrani** [lululxvi/deepxde#749] [Q&A]
Hi Dr. Lu,

I am developing a PINNs code to solve an inverse problem and I have a PDE in which the exponent is a float and not an integer of  dependent tensor as shown below in last term of eqn1:

del_stress = 420.0-5.0
stress_min = 5.0
del_time = 113.31-0.052
time_min = 0.052
del_shear_rate = 5.02-0.0797
shear_min = 0.0797
del_r = (14.455-13.334)*1e-3

eps = dde.Variable(0.5)

def pde(x,y):
    
    f = y[:,0:1]
    stress = y[:,1:2]
    
    shear_rate = x[:,0:1]
    r = x[:,1:2]
    t = x[:,2:3]
    
    df_dt = dde.grad.jacobian(y,x,i=0,j=2)
    d2f_dr2 = dde.grad.hessian(y,x,component=0,i=1,j=1)
    stress_rate = dde.grad.jacobian(y,x,i=1,j=2)
    
    # non-dimensional PDE #
    eqn1 = df_dt - f*del_time*(eps**2 *(d2f_dr2/del_r**2)  - f**1.5)  
    return eqn1
    
I have two questions:
 
1) Due to the float exponent (1.5 in this case), the model outputs a NaN value for PDE residual when I start training at step 0. But if I change the exponent from 1.5 to 2 in the last term, the model works fine. Can you please let me know why this is happening and if there is any workaround?

2) The dependent tensor f in above code is a function of independent tensors r and t. Is there a way to integrate this function f with respect to r so that it is only a function of t?

**Comment by lululxvi** [comment]
1. I guess it is due to that f is negative sometimes.
2. Might be doable, but not easy to do.

**Comment by Pranay-Nagrani** [comment]
For question 1: The tensor f has values between 0 and 1 from a physics perspective. Is there a way to make sure f does not go below 0? Further,  I tried to use tf.cast to convert the tensor to floating point type but it remains int32 always.

**Comment by lululxvi** [comment]
See the first example and code of https://doi.org/10.1137/21M1397908
